packages/flutter/test/services/text_boundary_test.dart

```diff
@@ -56,6 +56,18 @@ void main() {
     expect(boundary.getTrailingTextBoundaryAt(position).offset, TestTextLayoutMetrics.lineAt3.end);
   });
 
+  test('paragraph boundary works', () {
+    const String text = 'abcd efg hi\njklmno\npqrstuv';
+    const ParagraphBoundary boundary = ParagraphBoundary(text);
+
+    // Position enclosed inside of paragraph, 'abcd efg h|i\n'.
+    const TextPosition position = TextPosition(offset: 10);
+
+    // The range includes the line terminator.
+    expect(boundary.getLeadingTextBoundaryAt(position), const TextPosition(offset: 0));
+    expect(boundary.getTrailingTextBoundaryAt(position), const TextPosition(offset: 12, affinity: TextAffinity.upstream));
+  });
```

Comment:
Nit: If you think it would be useful, maybe test one or more edge cases of getTextBoundaryAt, like what happens if you pass it a series of newline characters.

Reply:
I think you've updated this to include a bunch of good edge cases now :+1: